docs/architecture/adr-047-handling-evidence-from-light-client.md

```diff
@@ -19,161 +20,200 @@ the light client detector receives two conflicting headers
+### Checking for a Lunatic Attack.
 
-* if `trustedHeader.Round == divergedHeader.Round`, and a validator signed for the block in both headers then DuplicateVoteEvidence can be immediately formed
+To quickly summarize a lunatic attack, the bisection verification works on the principle that if over 1/3
+of the validator power in the height I trusted has also voted for a future header I don't trust yet, 
+then from the Tendermint safety guarantee of 1/3, at least one validator in there must 
+not be faulty. This principle breaks down of course if there is a cabal anywhere within the trusted
+period that is greater than the light clients trust level (which is default at 1/3). This means that
+such a cabal can fool the light client into verifying almost any arbitrary header. The detector
+is predominantly designed as a second layer to detect such an attack.
 
-* if `trustedHeader.Round != divergedHeader.Round` then we form PotentialAmnesiaEvidence as some validators in this set have behaved maliciously and protocol in ADR 56 needs to be run. 
+Lunatic evidence has the following data structure:
 
-*The node does not check that there is a 1/3 overlap between headers as this may not be point of the fork and validator sets may have since changed*
+```go
+type LunaticAttackEvidence struct {
+	SignedHeader          *SignedHeader
+	ValidatorSourceHeight int64
+	Timestamp             time.Time
+}
+```
 
-If no evidence can be formed from a light trace, it is not a legitimate trace and thus the 
-connection with the peer should be stopped
+Notice that unlike `DuplicateVoteEvidence` this evidence groups all malicious validators together. 
+Another point of note is that the height and time, which is critical for calculating expiration is
+the header that the light client trusted not the header where the attack actually happened. This is
+done because after the header that the light client trusted we can't be sure that the validators
+are still in the set and hence are still bonded. 
 
-### F1. Equivocation
+`CheckForLunaticAttack(commonHeader, trustedHeader, divergedHeader *SignedHeader, commonValSet *ValidatorSet) (*Evidence, error)`
 
-Existing `DuplicateVoteEvidence` needs to be created and gossiped.
+A lunatic attack is leveraged from the common trusted header. This is the intermediary header that 
+came directly before the divergent headers and must contain a 1/3 overlap of validators.
 
+The two divergent headers must have at least one of the deterministically derived header fields different
+from one another. This could be one of `ValidatorsHash`, `NextValidatorsHash`, `ConsensusHash`,
+`AppHash`, and `LastResultsHash`. If all these fields are the same then the divergence is not a product
+of a lunatic attack. If so then we can generate either one or two `LunaticAttackEvidence` depending on the bifurcation point:
 
-### F5. Lunatic validator
+**IMPORTANT:** The light client cannot verify which validator set is the correct one 
```

Comment:
As we don't know what peer is correct, we should probably always work under assumption that we fix a trace (suspected) and assume that is a faulty, and then work with other peer to find a verifiable bifurkation point. If we succeed, then we only need to send information from the suspected trace to a peer, as we know that he has all needed information to verify it. Therefore, once we detect conflicting headers (one from primary and one from witness) we need to run detection procedure twice, once for each suspected trace. Note that if a peer is faulty and not collaborating we might fail to build evidences for both peers.